Gems/PhysX/Code/Tests/RigidBodyComponentTests.cpp

```diff
@@ -140,4 +140,65 @@ namespace PhysXEditorTests
         // Check the geometry is a type of Convex
         EXPECT_EQ(pxShape->getGeometryType(), physx::PxGeometryType::eCONVEXMESH);
     }
+
+    TEST_F(PhysXEditorFixture, EditorRigidBodyComponent_CylinderColliderZeroRadius_NoColliderCreated)
+    {
+        // Create editor entities
+        EntityPtr editorEntity = CreateInactiveEditorEntity("ZeroRadius");
+
+        UnitTest::ErrorHandler expectedError("SetCylinderRadius: radius must be greater than zero.");
+
+        const auto* rigidBodyComponent = editorEntity->CreateComponent<PhysX::EditorRigidBodyComponent>();
+        const auto* colliderComponent = editorEntity->CreateComponent<PhysX::EditorColliderComponent>();
+
+        editorEntity->Activate();
+
+        AZ::EntityComponentIdPair idPair(editorEntity->GetId(), colliderComponent->GetId());
+
+        // Set collider to be a cylinder
+        const Physics::ShapeType shapeType = Physics::ShapeType::Cylinder;
+        PhysX::EditorColliderComponentRequestBus::Event(idPair, &PhysX::EditorColliderComponentRequests::SetShapeType, shapeType);
+
+        // Set collider cylinder radius to zero
+        PhysX::EditorColliderComponentRequestBus::Event(idPair, &PhysX::EditorColliderComponentRequests::SetCylinderRadius, 0.0f);
+
+        // Notify listeners that collider has changed
+        Physics::ColliderComponentEventBus::Event(editorEntity->GetId(), &Physics::ColliderComponentEvents::OnColliderChanged);
```

Comment:
Is it not a bit weird having to do this manually? Or is it just because it's in a test and something that would normally handle this is not stood up?

Reply:
We get these automatically when working with colliders via UI in the Editor. 
Here `OnColliderChanged` is an indication that we finished making changes to the collider and RigidBodyComponent can re-create the actual rigid body. 
Doing it on every single change (i.e. inside SetCylinderRadius) wouldn't be great either as we can have incomplete setups used for constructing the rigid body which could spawn unnecessary warnings/errors..

Reply:
Cool, thanks for the explanation.